app/bundles/CoreBundle/Controller/Api/FileApiController.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/*
+ * @copyright   2016 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CoreBundle\Controller\Api;
+
+use Mautic\ApiBundle\Controller\CommonApiController;
+use Mautic\CoreBundle\Helper\InputHelper;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+
+/**
+ * Class FileApiController.
+ */
+class FileApiController extends CommonApiController
+{
+    public function initialize(FilterControllerEvent $event)
+    {
+        parent::initialize($event);
+        $this->entityNameOne   = 'file';
+        $this->entityNameMulti = 'files';
+    }
+
+    protected $forbiddenExtensions = [
+        'php',
+    ];
+
+    /**
+     * Uploads a file.
+     *
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function createAction($dir)
+    {
+        try {
+            $path = $this->getAbsolutePath($dir, true);
+        } catch (\Exception $e) {
+            return $this->returnError($e->getMessage(), Response::HTTP_NOT_ACCEPTABLE);
+        }
+
+        $response = [$this->entityNameOne => []];
+        if ($this->request->files) {
+            foreach ($this->request->files as $file) {
+                if (!in_array($file->guessExtension(), $this->forbiddenExtensions)) {
+                    $fileName = md5(uniqid()).'.'.$file->guessExtension();
+                    $moved    = $file->move($path, $fileName);
+
+                    if (substr($dir, 0, 6) === 'images') {
+                        $response[$this->entityNameOne]['link'] = $this->getMediaUrl().'/'.$fileName;
+                    }
+
+                    $response[$this->entityNameOne]['name'] = $fileName;
+                } else {
+                    return $this->returnError('The uploaded file cannot have these exetnsions: '.implode(',', $this->forbiddenExtensions).'.', Response::HTTP_NOT_ACCEPTABLE);
```

Comment:
I'll correct the spelling, allowed_extensions can be used I suppose. My goal was to disallow uploading executable scripts.

I'm not sure about the translations. If I'd implement API to for example French Mautic, I wouldn't want to get error messages in French. English is universal programming language.